swift/extractor/SwiftExtractor.cpp

```diff
@@ -124,6 +126,17 @@ static void extractDeclarations(const SwiftExtractorConfiguration& config,
 
 void codeql::extractSwiftFiles(const SwiftExtractorConfiguration& config,
                                swift::CompilerInstance& compiler) {
+  // The frontend can be called in many different ways.
+  // At each invocation we only extract system and builtin modules and any input source files that
+  // have an output associated with them.
+  std::unordered_set<std::string> sourceFiles;
+  auto inputFiles = compiler.getInvocation().getFrontendOptions().InputsAndOutputs.getAllInputs();
+  for (auto& input : inputFiles) {
+    if (input.getType() == swift::file_types::TY_Swift && !input.outputFilename().empty()) {
+      sourceFiles.insert(input.getFileName());
+    }
+  }
```

Comment:
I'm happy to do it in a follow up PR 😄

Reply:
hmm, if you are planning to touch this any way very soon, ok, otherwise I'm more of a "why not now?" guy when it comes to small refactorings aiding readability.

Reply:
I'm knee deep in some other task right now and wan't be able to apply the suggested change right now. OTOH, merging this earlier unblocks DCA for the Wire app, which is IMO is more important than a cosmetic change 😄